### PR TITLE
Add sanity check for CFF metadata

### PR DESCRIPTION
**Pull request by sdruskat**
This PR adds a sanity check to see if the metadata dict read from a CFF file contains at least the required CFF keys.

This is a first step towards complete validation (e.g., via `cffconvert --validate`).

### CODE REVIEW

File: code/analysis/python/python/read_cff_files.py [hunk 1]
```diff
@@ -19,14 +28,17 @@ def read_cff_files(datadir):
...
                     cff_file = yaml.safe_load(f)
-                    cff_data.append(cff_file)
+                    if sanity_check(cff_file):
+                        _cff_data.append(cff_file)
+                    else:
+                        _invalid_file.append(file)
                 except:
-                    invalid_file.append(file)
-    return cff_data, invalid_file
+                    _invalid_file.append(file)
```
Comment (markbasham) [inline code/analysis/python/python/read_cff_files.py]: Is it worth separating files which do not have the correct information, from ones where an exception is called?

Comment (sdruskat) [reply code/analysis/python/python/read_cff_files.py]: Yes, agree. I'll adapt.

Comment (sdruskat) [reply code/analysis/python/python/read_cff_files.py]: Should be fixed in 70ebf99674d24acf4b123712005199edb7cece5d.

**Review by markbasham — APPROVED** [review]
Looks good to me

**Review by markbasham — APPROVED** [review]
All good

**Pull request merged**